src/http/two/BinaryPacker.js

```diff
@@ -0,0 +1,119 @@
+import { Must } from "../../misc/Gadgets";
+
+export default class BinaryPacker {
+    constructor() {
+        this._data = new Buffer("", "binary");
+    }
+
+    binToHex(text) {
+        return Buffer(text, "binary").toString("hex");
+    }
+
+    raw() {
+        return this._data.toString("binary");
+    }
+
+    consume(length) {
+        let buf = new Buffer("", "binary");
+        Must(this._data.copy(buf, 0, 0, length) === length);
+        this._data = buf;
+    }
+
+    in(data, desc) {
+        this._data += data;
+        this.put(data, data.length, desc);
+    }
+
+    put(value, size, desc) {
+        console.log(`${desc} = ${typeof value === "number" ? value : this.binToHex(value)};`, size, "bytes long");
+    }
+
+    area(value, size, desc) {
```

Comment:
s/area/uint/ because you are writing an integer, not an opaque value

use this.in() instead of reimplementing it